server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -79,19 +82,34 @@ public ClientQuerySegmentWalker(
   private <T> QueryRunner<T> makeRunner(final Query<T> query)
   {
     final QueryToolChest<T, Query<T>> toolChest = warehouse.getToolChest(query);
-    final FinalizeResultsQueryRunner<T> baseRunner = new FinalizeResultsQueryRunner<T>(
-        toolChest.postMergeQueryDecoration(
-            toolChest.mergeResults(
-                toolChest.preMergeQueryDecoration(
-                    new RetryQueryRunner<T>(
+    final QueryRunner<T> baseRunner = CPUTimeMetricQueryRunner.safeBuild(
+        new FinalizeResultsQueryRunner<T>(
+            toolChest.postMergeQueryDecoration(
+                toolChest.mergeResults(
+                    toolChest.preMergeQueryDecoration(
+                        new RetryQueryRunner<T>(
                             baseClient,
                             toolChest,
                             retryConfig,
-                            objectMapper)
+                            objectMapper
+                        )
+                    )
                 )
-            )
+            ),
+            toolChest
         ),
-        toolChest
+        new Function<Query<T>, ServiceMetricEvent.Builder>()
+        {
+          @Nullable
+          @Override
+          public ServiceMetricEvent.Builder apply(Query<T> tQuery)
+          {
+            return toolChest.makeMetricBuilder(tQuery);
+          }
+        },
+        emitter,
+        new AtomicLong(0L),
```

Comment:
I assume you mean the atomiclong(0). Yes, it goes with the reporting boolean beneath it. If the runner is set to report, then it reports whatever the value of the atomiclong accumulator. It is entirely possible to have multiple runners use the same accumulator and only report once (but that is not the case here)